src/test/java/org/apache/drillui/test/framework/testng/EventsListener.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to you under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.drillui.test.framework.testng;
+
+import org.apache.commons.lang3.reflect.FieldUtils;
+import org.apache.drillui.test.framework.pages.BasePage;
+import org.apache.drillui.test.framework.pages.ErrorPage;
+import org.apache.drillui.test.framework.pages.QueryExceptionPage;
+import org.testng.ITestContext;
+import org.testng.ITestListener;
+import org.testng.ITestResult;
+
+public class EventsListener implements ITestListener {
```

Comment:
Can we use more clear name for the listener?

Reply:
Renamed to TestEventsListener